.github/workflows/build-images-base.yaml

```diff
@@ -65,6 +65,11 @@ jobs:
         run: |
           sudo apt-get update
           sudo apt-get install -y qemu-user-static
+      - name: Add git sha tag for the main branch
+        if: ${{ github.ref_name == env.MAIN_BRANCH_NAME }}
+        id: add-git-sha-tag
+        run: |
+          echo "IMG_TAGS=${{ github.sha }} ${{ env.IMG_TAGS }}" >> $GITHUB_ENV
```

Comment:
IMO, we probably want to keep [Build images for main branch](https://github.com/Kuadrant/limitador-operator/blob/main/.github/workflows/build-images-main.yaml) and get rid of [Schedule build with latest image SHA versions](https://github.com/Kuadrant/limitador-operator/blob/main/.github/workflows/build-images-branches.yaml). Maybe add `limitadorVersion` to the former?

Reply:
I was about to delete the scheduled build with the latest image with the git sha tag. But then I realized that this build is actually the correct one. The build triggered from push to main is not correct.

We need to fix this in a following up PR

That's good enough, @guicassolato ?

Reply:
IDK. I don't really like that main is built twice today. I would get rid of the [scheduled](https://github.com/Kuadrant/limitador-operator/blob/main/.github/workflows/build-images-scheduled.yaml) one right away and add `limitadorVersion: ${{ vars.LIMITADOR_SHA }}` to the [build main](https://github.com/Kuadrant/limitador-operator/blob/main/.github/workflows/build-images-main.yaml) one.

Reply:
Oh! Too soon. I see the problem with that now... When the operand changes, nothing will rebuild the operator with the updated SHA, right? Damn!

Reply:
So add `limitadorVersion: ${{ vars.LIMITADOR_SHA }}` to [build main](https://github.com/Kuadrant/limitador-operator/blob/main/.github/workflows/build-images-main.yaml) _and_ remove [this line](https://github.com/Kuadrant/limitador-operator/blob/248c5c9281d63ef30e9bf2deab07eddfa092dbea/.github/workflows/build-images-scheduled.yaml#L14) but keep both workflows? `main` will build twice but at least both tags will be added and linked every time. Right?

Reply:
> Oh! Too soon. I see the problem with that now... When the operand changes, nothing will rebuild the operator with the updated SHA, right? Damn!

The other option is to trigger build on the operator project when the operand gets a new build. But I rather prefer the nightly build approach instead. To avoid too many operator builds in the same day when the operand changes many times in the same day.

Reply:
I would like to fix that in another PR